p2p/shared_udp_connection.go

```diff
@@ -0,0 +1,84 @@
+package p2p
+
+import (
+	"errors"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"net"
+)
+
+// UDPReader implements a shared connection. Write sends messages to the
+// underlying connection while read returns messages that were found
+// unprocessable and sent to the unhandled channel by the primary listener.
+type UDPReader interface {
```

Comment:
This interface has number of methods that deal with UDP including read/write/close. It's deserved a more descriptive name than just a `Reader`.  And it seems a dup of local interface `conn` from `p2p/discover` package.

Is there a chance to move some to `discover` package or expose from `discover` package instead?

Reply:
Agree about the name.
Do you think I should export the `conn` interface from `discover` instead?